Big decimal overflow when converting to native

**Describe the bug**

Error occurred while running tpcds (q98 https://github.com/wForget/benchmarks-spark-native/actions/runs/12884500659/job/35921021019):

```
2025-01-21T10:11:55.2272588Z 25/01/21 10:11:55 INFO DAGScheduler: Job 2262 finished: collect at BenchmarkRun.scala:23, took 0.038236 s
2025-01-21T10:11:55.2343745Z ##[error]Exception in thread "main" java.lang.ArithmeticException: BigInteger out of long range
2025-01-21T10:11:55.2363964Z    at java.math.BigInteger.longValueExact(BigInteger.java:4640)
2025-01-21T10:11:55.2364370Z    at org.apache.spark.sql.types.Decimal.toUnscaledLong(Decimal.scala:221)
2025-01-21T10:11:55.2365081Z    at org.apache.spark.sql.blaze.NativeConverters$.convertValue(NativeConverters.scala:245)
2025-01-21T10:11:55.2365850Z    at org.apache.spark.sql.blaze.NativeConverters$.$anonfun$convertValue$1(NativeConverters.scala:255)
2025-01-21T10:11:55.2366429Z    at org.apache.spark.sql.blaze.NativeConverters$.$anonfun$convertValue$1$adapted(NativeConverters.scala:254)
2025-01-21T10:11:55.2366843Z    at org.apache.spark.sql.catalyst.util.ArrayData.foreach(ArrayData.scala:185)
2025-01-21T10:11:55.2367316Z    at org.apache.spark.sql.blaze.NativeConverters$.convertValue(NativeConverters.scala:254)
2025-01-21T10:11:55.2368270Z    at org.apache.spark.sql.execution.blaze.plan.NativeShuffleExchangeBase.$anonfun$prepareNativeShuffleDependency$6(NativeShuffleExchangeBase.scala:234)
2025-01-21T10:11:55.2368687Z    at scala.collection.TraversableLike.$anonfun$map$1(TraversableLike.scala:286)
2025-01-21T10:11:55.2369089Z    at scala.collection.mutable.ResizableArray.foreach(ResizableArray.scala:62)
2025-01-21T10:11:55.2369507Z    at scala.collection.mutable.ResizableArray.foreach$(ResizableArray.scala:55)
2025-01-21T10:11:55.2369890Z    at scala.collection.mutable.ArrayBuffer.foreach(ArrayBuffer.scala:49)
2025-01-21T10:11:55.2370208Z    at scala.collection.TraversableLike.map(TraversableLike.scala:286)
2025-01-21T10:11:55.2370507Z    at scala.collection.TraversableLike.map$(TraversableLike.scala:279)
2025-01-21T10:11:55.2370792Z    at scala.collection.AbstractTraversable.map(Traversable.scala:108)
2025-01-21T10:11:55.2371708Z    at org.apache.spark.sql.execution.blaze.plan.NativeShuffleExchangeBase.prepareNativeShuffleDependency(NativeShuffleExchangeBase.scala:229)
2025-01-21T10:11:55.2372609Z    at org.apache.spark.sql.execution.blaze.plan.NativeShuffleExchangeBase.shuffleDependency$lzycompute(NativeShuffleExchangeBase.scala:81)
2025-01-21T10:11:55.2373405Z    at org.apache.spark.sql.execution.blaze.plan.NativeShuffleExchangeBase.shuffleDependency(NativeShuffleExchangeBase.scala:75)
2025-01-21T10:11:55.2374374Z    at org.apache.spark.sql.execution.blaze.plan.NativeShuffleExchangeExec.mapOutputStatisticsFuture$lzycompute(NativeShuffleExchangeExec.scala:83)
2025-01-21T10:11:55.2375501Z    at org.apache.spark.sql.execution.blaze.plan.NativeShuffleExchangeExec.mapOutputStatisticsFuture(NativeShuffleExchangeExec.scala:78)
2025-01-21T10:11:55.2376226Z    at org.apache.spark.sql.execution.exchange.ShuffleExchangeLike.$anonfun$submitShuffleJob$1(ShuffleExchangeExec.scala:73)
2025-01-21T10:11:55.2376699Z    at org.apache.spark.sql.execution.SparkPlan.$anonfun$executeQuery$1(SparkPlan.scala:246)
2025-01-21T10:11:55.2377300Z    at org.apache.spark.rdd.RDDOperationScope$.withScope(RDDOperationScope.scala:151)
2025-01-21T10:11:55.2377847Z    at org.apache.spark.sql.execution.SparkPlan.executeQuery(SparkPlan.scala:243)
2025-01-21T10:11:55.2378536Z    at org.apache.spark.sql.execution.exchange.ShuffleExchangeLike.submitShuffleJob(ShuffleExchangeExec.scala:73)
2025-01-21T10:11:55.2379350Z    at org.apache.spark.sql.execution.exchange.ShuffleExchangeLike.submitShuffleJob$(ShuffleExchangeExec.scala:72)
```

**To Reproduce**



**Expected behavior**


**Screenshots**


**Additional context**

--- a/spark-extension/src/main/scala/org/apache/spark/sql/blaze/NativeConverters.scala
+++ b/spark-extension/src/main/scala/org/apache/spark/sql/blaze/NativeConverters.scala
@@ -123,7 +123,10 @@ object NativeConverters extends Logging {
       case NullType | BooleanType | ByteType | ShortType | IntegerType | LongType | FloatType |
           DoubleType | StringType | DateType | TimestampType =>
         true
-      case _: DecimalType => true
+      case t: DecimalType  if DecimalType.is64BitDecimalType(t) => true
+      case _: DecimalType =>
+        // blaze only supports 64-bit decimal
+        false
       case at: ArrayType => scalarTypeSupported(at.elementType)
       case m: MapType =>
         scalarTypeSupported(m.keyType) && scalarTypeSupported(m.valueType)

--- a/spark-extension/src/main/scala/org/apache/spark/sql/blaze/NativeConverters.scala
+++ b/spark-extension/src/main/scala/org/apache/spark/sql/blaze/NativeConverters.scala
@@ -123,7 +123,7 @@ object NativeConverters extends Logging {
       case NullType | BooleanType | ByteType | ShortType | IntegerType | LongType | FloatType |
           DoubleType | StringType | DateType | TimestampType =>
         true
-      case t: DecimalType  if DecimalType.is64BitDecimalType(t) => true
+      case t: DecimalType if DecimalType.is64BitDecimalType(t) => true
       case _: DecimalType =>
         // blaze only supports 64-bit decimal
         false

